applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_2D2N.cpp

```diff
@@ -806,6 +806,98 @@ double CrBeamElement2D2N::Modulus2Pi(double A) {
   KRATOS_CATCH("")
 }
 
+void CrBeamElement2D2N::AddExplicitContribution(
+    const VectorType &rRHSVector, const Variable<VectorType> &rRHSVariable,
+    Variable<array_1d<double, 3>> &rDestinationVariable,
+    const ProcessInfo &rCurrentProcessInfo) {
+  // FORCE- & Moment- Residual is 3D vector
+  KRATOS_TRY;
+
+  bounded_vector<double, msElementSize> damping_residual_contribution =
+      ZeroVector(msElementSize);
+  // calculate damping contribution to residual -->
+  if ((this->GetProperties().Has(RAYLEIGH_ALPHA) ||
+       this->GetProperties().Has(RAYLEIGH_BETA)) &&
+      (rDestinationVariable != NODAL_INERTIA)) {
+    Vector current_nodal_velocities = ZeroVector(msElementSize);
+    this->GetFirstDerivativesVector(current_nodal_velocities);
+    Matrix damping_matrix = ZeroMatrix(msElementSize, msElementSize);
+    ProcessInfo temp_process_information; // cant pass const ProcessInfo
+    this->CalculateDampingMatrix(damping_matrix, temp_process_information);
+    // current residual contribution due to damping
+    damping_residual_contribution =
+        prod(damping_matrix, current_nodal_velocities);
+  }
+
+  if (rRHSVariable == RESIDUAL_VECTOR &&
+      rDestinationVariable == FORCE_RESIDUAL) {
+
+    for (SizeType i = 0; i < msNumberOfNodes; ++i) {
+      SizeType index = msLocalSize * i;
+
+      GetGeometry()[i].SetLock();
+
+      array_1d<double, 3> &r_force_residual =
+          GetGeometry()[i].FastGetSolutionStepValue(FORCE_RESIDUAL);
+
+      for (SizeType j = 0; j < msDimension; ++j) {
+        r_force_residual[j] +=
+            rRHSVector[index + j] - damping_residual_contribution[index + j];
+      }
+
+      r_force_residual[msDimension] = 0.00;
+      GetGeometry()[i].UnSetLock();
+    }
+  }
+
+  if (rRHSVariable == RESIDUAL_VECTOR &&
+      rDestinationVariable == MOMENT_RESIDUAL) {
+
+    for (SizeType i = 0; i < msNumberOfNodes; ++i) {
+      SizeType index = (msLocalSize * i) + msDimension;
+
+      GetGeometry()[i].SetLock();
+
+      array_1d<double, 3> &r_moment_residual =
+          GetGeometry()[i].FastGetSolutionStepValue(MOMENT_RESIDUAL);
+
+      for (SizeType j = 0; j < msDimension; ++j) {
+        r_moment_residual[j] = 0.00;
+      }
+      r_moment_residual[msDimension] +=
+          rRHSVector[index] - damping_residual_contribution[index];
+      GetGeometry()[i].UnSetLock();
+    }
+  }
+
+  if (rDestinationVariable == NODAL_INERTIA) {
+
+    Matrix element_mass_matrix = ZeroMatrix(msElementSize, msElementSize);
+    ProcessInfo temp_info; // Dummy

```

Comment:
why do you create a dummy? I think you have the real processinfo available

Reply:
:+1: